# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/nsHttpConnection.h
# Commit: 18651e373837
# Full Hash: 18651e373837007e879e22c9573f266698752a9a
# Author: Dragana Damjanovic <dd.mozilla@gmail.com>
# Date: 2021-09-22 15:53:06
# Regressor Bug: 1382886
# File Overlap Count: 1
# Description:
#   Bug 1382886 - Make sure that nsHttpConnection immediately knows that the handshake is done. r=necko-reviewers,kershaw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D123919
# ==============================================================================

diff -r f8d31e3ebc93 -r 18651e373837 netwerk/protocol/http/nsHttpConnection.h
--- a/netwerk/protocol/http/nsHttpConnection.h	Wed Sep 22 09:19:31 2021 +0000
+++ b/netwerk/protocol/http/nsHttpConnection.h	Wed Sep 22 09:19:31 2021 +0000
@@ -249,6 +249,7 @@
   void EarlyDataTelemetry(int16_t tlsVersion, bool earlyDataAccepted);
   void FinishNPNSetup(bool handshakeSucceeded, bool hasSecurityInfo);
   void Reset0RttForSpdy();
+  void HandshakeDoneInternal();
 
  private:
   // mTransaction only points to the HTTP Transaction callbacks if the
@@ -358,6 +359,7 @@
   bool mEarlyDataNegotiated{false};  // Only used for telemetry
   nsCString mEarlyNegotiatedALPN;
   bool mDid0RTTSpdy{false};
+  bool mTlsHandshakeComplitionPending{false};
 
   nsresult mErrorBeforeConnect = NS_OK;
 